### DISCUSSION TITLE
Log Forging with types different than string.

### DISCUSSION THREAD

**Discussion by WSkwieVolue** [github/codeql#10454] [Q&A] [answered]
https://github.com/github/codeql/blob/bc93a22e7de7192c6f316f8262a528689edd1766/csharp/ql/src/Security%20Features/CWE-117/LogForging.ql#L1

Hello :),

I would like to know why does this rule apply for parameters with types different than string. For example I have received an alert with high severity about a DateTime variable. It is formatted like this:

`{validFor:yyyy-MM-dd}`

So there is no way of a malicious user to forge anything this way. I do not see any other way to sanitize that input other than formatting it.

Could you please explain this to me?

**Comment by edoardopirovano** [comment, marked as answer]
Greetings, many thanks for raising this with us. It does indeed sound like there is an opportunity to reduce false positives on this query by enhancing our analysis to consider the types of variables flowing into the logs. I'll raise this with our C# analysis team so they can consider whether to prioritize making this change.

**Reply by WSkwieVolue** [reply]
Thank You :)

**Comment by michaelnebel** [comment]
@WSkwieVolue : Could you provide a small piece of example code? Is the alert found in some ASP.NET code (you mention parameters)?

**Reply by WSkwieVolue** [reply]
In one of my controllers there is a endpoint method that takes a datetime as a parameter. That parameter is later passed to inner methods. The CodeQL sees that this datetime is used directly to format a log line in the inner methods.

I cannot share my exact code but it is something similar to this:

```
[HttpGet]
public async Task<ActionResult> GetSomething(DateTime date)
{
    SomeMethod(date);
}

private SomeMethod(DateTime date)
{
   (...)
    _Logger.Error($"Could not do stuff for selected date {date:yyyy-MM-dd}")
}
```

**Reply by michaelnebel** [reply]
Great - thank you! I expected something along those lines :-)
